src/routes/assessmentRoutes.js

```diff
@@ -71,84 +106,139 @@ router.get("/assessments/:studentId", isLoggedIn, async (req, res) => {
 
-      // Fetch evaluations for each student without populating evaluator
+      // Fetch student details for those evaluatees
+      const students = await User.find({
+        _id: { $in: studentIds },
+        user_type: "student",
+      });
+
+      // Fetch evaluations for each student
       const studentsWithEvaluations = await Promise.all(
         students.map(async (student) => {
-          const evaluations = await Evaluation.find({ evaluatee: student._id });
-          return { student, evaluations };
+          const studentEvaluations = evaluations.filter(
+            (eval) => eval.evaluatee.toString() === student._id.toString(),
+          );
+          return { student, evaluations: studentEvaluations };
         }),
       );
 
-      // Render the page with the students and evaluations
-      res.render("assessments_detailed", { studentsWithEvaluations });
+      res.render("assessments_detailed", { studentsWithEvaluations, classId });
     } catch (e) {
+      console.error(e);
       res.status(500).json({ error: e.message });
     }
   },
 );
 
 //Summarized Assessments
 router.get(
-  "/assessments-summary",
+  "/assessments-summary/:classId",
   isLoggedIn,
   isInstructor,
   async (req, res) => {
     try {
-      //Aggregate the average ratings and scores for each student
-      const summary = await Evaluation.aggregate([
-        {
-          $group: {
-            _id: "$evaluatee", // Group by evaluatee (student)
-            averageCooperation: { $avg: "$cooperation.rating" },
-            averageConceptual: { $avg: "$conceptual_contribution.rating" },
-            averagePractical: { $avg: "$practical_contribution.rating" },
-            averageWorkEthic: { $avg: "$work_ethic.rating" },
-            overallAverageScore: { $avg: "$average_score" },
-          },
-        },
-      ]);
-
-      //Populate student data for the summary
-      const populatedSummary = await Promise.all(
-        summary.map(async (record) => {
-          const student = await User.findById(record._id);
-          return {
+      const { classId } = req.params;
+
+      // Validate the class exists
+      const currentClass = await Class.findById(classId);
+      if (!currentClass) {
+        return res.status(404).send("Class not found");
+      }
+
+      // Ensure the class is linked to the current instructor
+      if (!currentClass.instructor_id.equals(req.user._id)) {
+        return res.status(403).send("You are not authorized to view this class.");
+      }
+
+      // Fetch unique evaluatee IDs from evaluations for the class
+      const evaluations = await Evaluation.find({ class_id: classId });
+      const studentIds = [...new Set(evaluations.map((eval) => eval.evaluatee))];
+
+      // Fetch detailed student records
+      const students = await User.find({
+        _id: { $in: studentIds },
+        user_type: "student",
+      });
+
+      // Aggregate average ratings and scores for each student
```

Comment:
Handle cases where totals.count is 0 to avoid potential division by zero errors.